ChemistryLib/Output.h

```diff
@@ -0,0 +1,104 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <fstream>
+#include <string>
+#include <vector>
+
+namespace ChemistryLib
+{
+class BasicOutputSetups
+{
+public:
+    static inline const std::string output_file = "phreeqc.out";
+
+private:
+    static const bool display_simulation_id = false;
+    static const bool display_state = true;
+    static const bool display_solution_id = true;
+    static const bool display_distance = false;
+    static const bool display_current_time = false;
+    static const bool display_time_step = false;
+    static const bool display_pH = true;
+    static const bool display_pe = true;
+    static const bool use_high_precision = true;
+
+public:
+    int getNumberOfItemsInDisplay()
+    {
+        return display_simulation_id + display_state + display_solution_id +
+               display_distance + display_current_time + display_time_step +
+               display_pH + display_pe;
+    }
+
+    int getNumberOfDroppedItems()
+    {
+        return display_simulation_id + display_state + display_solution_id +
+               display_distance + display_current_time + display_time_step;
+    }
+
+    friend std::ofstream& operator<<(
+        std::ofstream& out, BasicOutputSetups const& basic_output_setups);
+};
+
+enum class ItemType
+{
+    pH,
+    pe,
+    Component,
+    EquilibriumPhase,
+    KineticReactant
+};
+
+struct OutputItem
+{
+    OutputItem(std::string name_, ItemType item_type_)
+        : name(std::move(name_)), item_type(item_type_)
+    {
+    }
+
+    std::string const name;
+    ItemType const item_type;
+};
+
+struct Output
+{
+    Output(BasicOutputSetups basic_output_setups_,
+           std::vector<OutputItem>
```

Comment:
`std::vector<OutputItem>` --> `std::vector<OutputItem>&&`
` std::vector<int>` --> ` std::vector<int>&&`
With the above changes, add std::move to the variables  of the corresponding argument of Output.

Reply:
Changed.

Reply:
It seems that your changes have not yet committed.